src/vivarium_public_health/disease/special_disease.py

```diff
@@ -104,24 +108,38 @@ def setup(self, builder):
         threshold = builder.configuration[self.cause.name].threshold
 
         self.filter_by_exposure = self.get_exposure_filter(distribution, exposure_pipeline, threshold)
-        self.population_view = builder.population.get_view([self.cause.name, 'alive'])
+        self.population_view = builder.population.get_view([self.cause.name, self.diseased_event_time_column,
+                                                            self.susceptible_event_time_column, 'alive'])
 
         builder.event.register_listener('time_step', self.on_time_step)
-        builder.population.initializes_simulants(self.on_initialize_simulants)
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 creates_columns=[self.cause.name,
+                                                                  self.diseased_event_time_column,
+                                                                  self.susceptible_event_time_column])
 
     def on_initialize_simulants(self, pop_data):
-        new_pop = pd.Series(f'susceptible_to_{self.cause.name}', index=pop_data.index, name=self.cause.name)
+        new_pop = pd.DataFrame({self.cause.name: f'susceptible_to_{self.cause.name}',
+                                self.diseased_event_time_column: pd.Series(pd.NaT, index=pop_data.index),
+                                self.susceptible_event_time_column: pd.Series(pd.NaT, index=pop_data.index)},
+                               index=pop_data.index)
+
         sick = self.filter_by_exposure(pop_data.index)
-        new_pop[sick] = self.cause.name
+        new_pop.loc[sick, self.cause.name] = self.cause.name
+        new_pop.loc[sick, self.diseased_event_time_column] = self.clock()  # match VPH disease, only set w/ condition
         self.population_view.update(new_pop)
 
     def on_time_step(self, event):
         pop = self.population_view.get(event.index, query='alive == "alive"')
         sick = self.filter_by_exposure(pop.index)
         #  if this is recoverable, anyone who gets lower exposure in the event goes back in to susceptible status.
         if self.recoverable:
+            change_to_susceptible = (~sick) & (pop[self.cause.name] != f'susceptible_to_{self.cause.name}')
+            pop.loc[change_to_susceptible, self.susceptible_event_time_column] = event.time
             pop.loc[~sick, self.cause.name] = f'susceptible_to_{self.cause.name}'
+        change_to_diseased = sick & (pop[self.cause.name] != self.cause.name)
+        pop.loc[change_to_diseased, self.diseased_event_time_column] = event.time
```

Comment:
it shouldn't matter in terms of what the state table actually looks like, but I think it might make it a bit more readable if you used the change_to_susceptible instead of ~sick and change_to_diseased instead of sick to update the self.cause.name columns as well. as it is, we update the disease column every time step but event time columns only if the state changes. It seems better to just do both only if the state changes.